lambda/simplif.ml

```diff
@@ -294,38 +301,38 @@ let simplify_exits lam =
       | Not_found -> Lstaticraise (i,ls)
       end
   | Lstaticcatch (l1,(i,[]),(Lstaticraise (_j,[]) as l2)) ->
-      Hashtbl.add subst i ([],simplif l2) ;
-      simplif l1
+      Hashtbl.add subst i ([],simplif depth l2) ;
+      simplif depth l1
   | Lstaticcatch (l1,(i,xs),l2) ->
       let {count; max_depth} = get_exit i in
       if count = 0 then
         (* Discard staticcatch: not matching exit *)
-        simplif l1
-      else if count = 1 && max_depth <= !try_depth then begin
+        simplif depth l1
+      else if 
+      count = 1 && max_depth <= depth then begin
         (* Inline handler if there is a single occurrence and it is not
            nested within an inner try..with *)
-        assert(max_depth = !try_depth);
-        Hashtbl.add subst i (xs,simplif l2);
-        simplif l1
+        assert(max_depth = depth);
+        Hashtbl.add subst i (xs,simplif depth l2);
+        simplif depth l1
       end else
-        Lstaticcatch (simplif l1, (i,xs), simplif l2)
+        Lstaticcatch (simplif depth l1, (i,xs), simplif depth l2)
   | Ltrywith(l1, v, l2) ->
-      incr try_depth;
-      let l1 = simplif l1 in
-      decr try_depth;
-      Ltrywith(l1, v, simplif l2)
-  | Lifthenelse(l1, l2, l3) -> Lifthenelse(simplif l1, simplif l2, simplif l3)
-  | Lsequence(l1, l2) -> Lsequence(simplif l1, simplif l2)
-  | Lwhile(l1, l2) -> Lwhile(simplif l1, simplif l2)
+      let depth1 = depth +1 in
+      let l1 = simplif depth1 l1 in
```

Comment:
Using `simplif (depth + 1) l1` and removing the `depth1` let-binding above probably better here.